packages/eas-cli/src/credentials/ios/actions/SetUpAdhocProvisioningProfile.ts

```diff
@@ -177,7 +177,23 @@ export class SetUpAdhocProvisioningProfile {
       );
     }
 
-    // 6. Create (or update) app build credentials
+    // 6. Compare selected devices with the ones actually provisioned
+    const diffList = differenceBy(
+      chosenDevices,
+      appleProvisioningProfile.appleDevices,
+      'identifier'
+    );
+    if (diffList && diffList.length > 0) {
+      Log.warn(`Failed to provision ${diffList.length} of the selected devices:`);
+      for (const missingDevice of diffList) {
+        Log.warn(`- ${formatDeviceLabel(missingDevice)}`);
+      }
+      Log.log(
+        'Most commonly devices fail to to be provisioned while they are still being processed by Apple, which can take up to 24-72 hours. Check your Apple Developer Portal > Certificates, Identifiers & Profiles > Devices page, the devices in "Processing" status cannot be provisioned yet'
```

Comment:
I _love_ this messaging. It's clear, it's actionable, it offers a best guess over being vague, just perfect!

Reply:
Updated the message with the link